lib/api/2.0/obms.js

```diff
@@ -45,13 +46,56 @@ var obmsGetById = controller(function(req) {
 
 // PATCH /api/2.0/obms/identifier
 var obmsPatchById = controller( function(req) {
-    return waterline.obms.updateByIdentifier(
-        req.swagger.params.identifier.value, req.swagger.params.body.value);
+    var obmId = req.swagger.params.identifier.value;
+    var values = req.swagger.params.body.value;
+    return waterline.obms.needByIdentifier(obmId)
+    .then(function(oldObm) {
+        if (oldObm.node && !values.nodeId) {
```

Comment:
Please add some comment for each `if` branch to help other developer to understand the code.